llvm/test/CodeGen/AMDGPU/iglp-no-clobber.ll

```diff
@@ -0,0 +1,39 @@
+; RUN: llc -mtriple=amdgcn -mcpu=gfx942 --stop-after=si-fix-sgpr-copies < %s | FileCheck %s
+
+; iglp.opt should not be flagged as clobbering the memory operand for the global_load, and we should be able to
+; lower into the scalar version (i.e. should not need to lower into vector version with waterfall loop)
+; CHECK-NOT: WATERFALL
```

Comment:
This is much too fragile of a check. Positively check the output, preferably the final output

Reply:
Ok - since theres a couple comments I'll open a separate PR

Reply:
https://github.com/llvm/llvm-project/pull/127141